lib/ansible/modules/cloud/amazon/iam_server_certificate_info.py

```diff
@@ -36,136 +42,166 @@
+                    sample: "2017-06-15T12:00:00+00:00"
+                upload_date:
+                    description: The date and time this server certificate was uploaded, in ISO 8601 format.
+                    returned: always
+                    type: str
+                    sample: "2015-04-25T00:36:40+00:00"
 '''
 
+from ansible.module_utils.aws.core import AnsibleAWSModule
+from ansible.module_utils.ec2 import AWSRetry, camel_dict_to_snake_dict
 
+# Non-ansible imports
 try:
-    import boto3
-    import botocore.exceptions
-    HAS_BOTO3 = True
+    import botocore
 except ImportError:
-    HAS_BOTO3 = False
+    pass  # caught by AnsibleAWSModule
 
-from ansible.module_utils.basic import AnsibleModule
-from ansible.module_utils.ec2 import boto3_conn, ec2_argument_spec, get_aws_connection_info
 
+@AWSRetry.backoff(tries=5, delay=5, backoff=2.0)
+def list_iam_server_certificates_with_backoff(connection, path):
+    paginator = connection.get_paginator('list_server_certificates')
+    if path:
+        return paginator.paginate(PathPrefix=path).build_full_result()
+    else:
+        return paginator.paginate().build_full_result()
 
-def get_server_certs(iam, name=None):
-    """Retrieve the attributes of a server certificate if it exists or all certs.
-    Args:
-        iam (botocore.client.IAM): The boto3 iam instance.
-
-    Kwargs:
-        name (str): The name of the server certificate.
-
-    Basic Usage:
-        >>> import boto3
-        >>> iam = boto3.client('iam')
-        >>> name = "server-cert-name"
-        >>> results = get_server_certs(iam, name)
-        {
-            "upload_date": "2015-04-25T00:36:40+00:00",
-            "server_certificate_id": "ADWAJXWTZAXIPIMQHMJPO",
-            "certificate_body": "-----BEGIN CERTIFICATE-----\nbunch of random data\n-----END CERTIFICATE-----",
-            "server_certificate_name": "server-cert-name",
-            "expiration": "2017-06-15T12:00:00+00:00",
-            "path": "/",
-            "arn": "arn:aws:iam::911277865346:server-certificate/server-cert-name"
-        }
+
+def _get_server_certs(connection, module, name, path):
     """
-    results = dict()
-    try:
-        if name:
-            server_certs = [iam.get_server_certificate(ServerCertificateName=name)['ServerCertificate']]
-        else:
-            server_certs = iam.list_server_certificates()['ServerCertificateMetadataList']
+    Get one or more IAM server certificates facts based on name or path. If not found, return None.
 
+    :param connection: AWS boto3 glue connection
+    :param module: Ansible module
+    :param name: The name of the server certificate
+    :param path: The path of the server certificate
+    :return: IAM server certificates dict or None if not found
+    """
+
+    response = dict()
+    response['certificates'] = []
+
+    if name:
+        try:
+            response['certificates'].append(connection.get_server_certificate(ServerCertificateName=name)['ServerCertificate'])
```

Comment:
Since you're already using the AWSRetry decorator, it's worth wrapping this call too.  (The big one iIt would catch is API Rate Limit errors.)
If you include the the NoSuchEntity handling, make sure you rethrow any other ClientError exceptions.  Avoid handling any other exceptions inside your wrapper as that would prevent the decorator from doing its job.